### PR TITLE
Fix BUGS and Tests

### LINKED ISSUE #357: Add tests
- Ingress validator
- Template renderer

### PR DESCRIPTION
**Pull request by sadlil**
- Fix Bugs and Tests for api-rev
- Fixed old unit tests
- Added Some unit tests, [more needs to be added]

Fixes #357

### CODE REVIEW

File: pkg/haproxy/renderer.go [hunk 1]
```diff
@@ -4,7 +4,7 @@ import (
 	"bytes"
 	"fmt"
 	"sort"
-
+	//"github.com/appscode/go/strings"
```
Comment (tamalsaha) [inline pkg/haproxy/renderer.go]: Delete this line.

File: pkg/haproxy/renderer.go [hunk 2]
```diff
@@ -20,7 +20,8 @@ func RenderConfig(data TemplateData) (string, error) {
 		log.Error(err)
```
Comment (tamalsaha) [inline pkg/haproxy/renderer.go]: Do we not print this err at the caller? If not, log there. If yes, remove logging here.

File: pkg/ingress/parser.go [hunk 3]
```diff
@@ -274,7 +275,7 @@ func (c *Controller) generateConfig() error {
 			if len(eps) > 0 {
 				def := &haproxy.TCPService{
 					SharedInfo:   si,
-					FrontendName: fmt.Sprintf("tcp-%s", rule.TCP.Port),
+					FrontendName: fmt.Sprintf("tcp-%s", rule.TCP.Port.String()),
```
Comment (tamalsaha) [inline pkg/ingress/parser.go]: Do we need to call `.String()` method? If yes, then `Ingress.IsValid()` method also needs to call String() in many places.

Comment (sadlil) [inline pkg/ingress/parser.go]: yes we do need to call String(), before when we are not calling `String()` the template generated have values like `*intstr{some random value}`

File: pkg/haproxy/renderer.go (hunk 2 above)
Comment (sadlil) [inline pkg/haproxy/renderer.go]: we added an event for ingress, didn't logged. so keeping a log will be good.

**Pull request merged**